### PR TITLE
Fix issue where 2 IDE analyzers are using IDE0035

### LINKED ISSUE #23983: Two IDE diagnostic ids use the same value.
public const string ValidateFormatStringDiagnosticID = "IDE0035";
        public const string RemoveUnreachableCodeDiagnosticId = "IDE0035";

### PR DESCRIPTION
**Pull request by chborl**
### Customer scenario
Both the Validate Format String and Remove Unreachable Code analyzers use the same ID, so a user would see the same code for both of these issues and if they suppress either one it would suppress both.

### Bugs this fixes
#23983 

### Workarounds, if any
none

### Risk
No code risk, but it is a breaking change.  If a customer previously suppressed the Validate Format String analzyer, it would start firing again because it has a new ID number.  @kuhlenh will change it in our documentation and add it to the breaking change log.

### Performance impact
none

### Is this a regression from a previous update?
no

### Test documentation updated?

</details>

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
@chborl Can we add a test that we don't have duplicates? Something that just uses reflection to grab all the const values in this type and makes sure there aren't any duplicates? It's not perfect, but would have caught this.

@jinujoseph I'm guessing we want this in 15.7?

**Comment by jinujoseph** [comment]
yes , lets take this for 15.7

**Comment by chborl** [comment]
@jasonmalinowski , I've added a test for duplicate IDs so hopefully this won't happen in the future.

File: src/EditorFeatures/Test/Diagnostics/IDEDiagnosticIDUniquenessTest.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.Diagnostics
+{
+    public class IDEDiagnosticIDUniquenessTest
+    {
+        [Fact, Trait(Traits.Feature, Traits.Features.Diagnostics)]
+        public void UniqueIDEDiagnosticIds()
+        {
+            Type type = typeof(IDEDiagnosticIds);
+            List<string> ListOfIDEDiagnosticIds = type.GetFields().Select(x => x.GetValue(null).ToString()).ToList();
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Test/Diagnostics/IDEDiagnosticIDUniquenessTest.cs]: nit: local, so lowercase 'l'.

Comment (chborl) [reply src/EditorFeatures/Test/Diagnostics/IDEDiagnosticIDUniquenessTest.cs]: Good catch, fixed.

**Comment by sharwell** [comment]
:memo: Someone needs to backport this to 15.7 since it was merged before retargeting.

**Comment by chborl** [comment]
Thanks, @sharwell!   I'll put this in 15.7 as well.
